### PR TITLE
libyang 1.0.73

### PR DESCRIPTION
**Pull request by li-pingmao**
- build libyang1.0.73 debian pacakge from libyang github source
    - build libyang python2 and python3 debian packages

The main reasons this change:

We need libyang python debian packages. libyang python2 debian package needed by sonic-utility CLI for dynamic port breakout.
Build libyang from source to pick up bug fixes

**- What I did**
    - Build libyang version 1.0.73 from libyang github source
    - Build libyang python2 and python3 debian packages

**- How I did it**
 - libyang Makefile change to pull libyang from github source
 - build libyang debian package from libyang source
 - patches for debian files and Makefiles to build both libyang python2 and python3 debian packages

**- How to verify it**
libyang debian packages built:

target/debs/stretch/libyang_1.0.73_amd64.deb 
target/debs/stretch/libyang-cpp_1.0.73_amd64.deb  
target/debs/stretch/libyang-dbg_1.0.73_amd64.deb  
target/debs/stretch/libyang-dev_1.0.73_amd64.deb  
target/debs/stretch/python2-yang_1.0.73_amd64.deb
target/debs/stretch/python3-yang_1.0.73_amd64.deb

libyang python yang.py is installed in both python2 and python3 after installing the packages: 

./usr/lib/python2.7/dist-packages/yang.py
./usr/lib/python3/dist-packages/yang.py

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Comment by zhenggen-xu** [comment]
retest this please

**Comment by pavel-shirshov** [comment]
Can you please put in description why do we need this change?

File: src/libyang/patch/libyang.patch [hunk 1]
```diff
@@ -0,0 +1,134 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index fa562dd3..8635ba15 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -33,6 +33,7 @@ set(LIBYANG_MICRO_SOVERSION 2)
+ set(LIBYANG_SOVERSION_FULL ${LIBYANG_MAJOR_SOVERSION}.${LIBYANG_MINOR_SOVERSION}.${LIBYANG_MICRO_SOVERSION})
+ set(LIBYANG_SOVERSION ${LIBYANG_MAJOR_SOVERSION})
+ 
++set(CMAKE_INSTALL_PREFIX /usr)
```
Comment (pavel-shirshov) [inline src/libyang/patch/libyang.patch]: I think it'd better to use cmake -DCMAKE_INSTALL_PREFIX=/usr instead

**Comment by li-pingmao** [comment]
The main reasons this change is needed:
- We need libyang python debian packages.  libyang python2 debian package needed by sonic-utility CLI for dynamic port breakout.
- Build libyang from source to pick up bug fixes

**Pull request merged**